lib/chef/resource/osx_profile.rb

```diff
@@ -0,0 +1,84 @@
+#
+# Author:: Nate Walck (<nate.walck@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource'
+
+class Chef
+  class Resource
+    class OsxProfile < Chef::Resource
+      provides :osx_profile, os: "darwin"
+      provides :osx_config_profile, os: "darwin"
+
+      identity_attr :profile_name
+
+      state_attrs :installed
+
+      default_action :install
+      allowed_actions :install, :remove
+
+      def initialize(name, run_context=nil)
+        super
+        @profile_name = name
+        @profile = nil
+        @identifier = nil
+        @path = nil
+      end
+
+      def profile_name(arg=nil)
+        set_or_return(
+          :profile_name,
+          arg,
+          :kind_of => [ String ]
+        )
+      end
+
+      def installed(arg=nil)
```

Comment:
Totally showing my newb status here - I see that this is a `state_attr` so it will be sent to the reporting server.  But it seems weird to have a set-able property that seems to me to be read_only.

Its been a while since I've written a non-provisioning resource - is this just a common pattern?